lib/elixir/lib/application.ex

```diff
@@ -4,32 +4,89 @@ defmodule Application do
+
+  You'll find the resource file of an application in its `ebin` directory, it is
+  generated automatically by Mix. Some of its keys are taken from the keyword
+  lists returned by the `project/0` and `application/0` functions defined in
+  `mix.exs`, and others are generated by Mix itself.
+
+  You can learn more about the generation of application resource files in the
+  [`mix compile.app`](../mix/Mix.Tasks.Compile.App.html) task.
+
+  ## The application environment
+
+  The key `env` of an application resource file has a list of tuples that map
+  atoms to terms, and its contents are known as the application *environment*.
+  Note that this environment is unrelated to the operating system environment
+  variables.
+
+  The application environment can be overriden via the `-config` option of
+  `erl`, as well as command-line flags, as we are going to see below.
+
+  ## Loading applications
+
+  Applications are *loaded*, which means that the BEAM finds and processes their
+  resource file:
+
+      iex(1)> Application.load(:ex_unit)
+      :ok
+
+  If an application has included applications, they are loaded too. And the
+  procedure recurses if they in turn have included applications. Included
+  applications are unrelated to applications in Mix umbrella projects, they are
+  an Erlang concept that has to do with coordinated starts.
+
+  When an application is loaded, the environment specified in its resource file
+  is merged with any overrides from config files passed to `erl` via the
+  `-config` option. It is worth highlighting that releases pass `sys.config`
+  this way. In turn, these are merged again with any specific `-Application`
+  flags passed to `erl`.
+
+  Loading an application *does not* load its modules.
+
+  In practice, you rarely load applications by hand because that is part of the
+  start process, explained next.
+
+  ## Starting and stopping applications
+
+  Applications are also *started* and *stopped*:
+
+      iex(1)> Application.start(:ex_unit)
+      :ok
+      iex(2)> Application.stop(:ex_unit)
+      :ok
 
   Once your application is compiled, running your system is a matter of starting
   your current application and its dependencies. Differently from other languages,
   Elixir does not have a `main` procedure that is responsible for starting your
   system. Instead, you start one or more applications, each with their own
   initialization and termination logic.
 
-  Applications also provide an "application environment", which provides one
-  mechanism for configuring long running applications. We will learn more about
-  the tooling, start and shutdown and the application environment in the next
-  sections.
+  When an application is started, the BEAM loads it if it is not already loaded
+  (in the technical sense described above). Then, it checks if the dependencies
+  listed in the `applications` key of the resource file are already started.
```

Comment:
I think application should be an atom  since its a key

Reply:
I thought about commenting the same, but if you open the application resource file `APP.app` what you find is `applications`, not `:applications`. So I thought the current documentation was more accurate in the sense of "what you will see if you open the file".

Reply:
That was also my rationale, because it is the ... key *of the resource file*.

Reply:
Yes. It should stay as is then. Sorry i didnt have a computer at hand to check the files when reviewing